apis/v1alpha1/httproute_types.go

```diff
@@ -78,6 +88,23 @@ type HTTPRouteHost struct {
 	ExtensionRef *RouteHostExtensionObjectReference `json:"extensionRef"`
 }
 
+// RouteTLSConfig describes a TLS configuration defined at the Route level.
+type RouteTLSConfig struct {
+	// CertificateRef is the reference to Kubernetes object that
+	// contain a TLS certificate and private key.
+	// This certificate MUST be used for TLS handshakes for the domain
+	// this RouteTLSConfig is associated with.
```

Comment:
What if this domain is used by other routes that provide other certificates? This could potentially be interpreted as: "any TLS handshake involving the domain defined here should use this cert".

Reply:
> What if this domain is used by other routes that provide other certificates?

That is a conflict. Route can't be selected at the time of TLS session so only one route should define a TLS certificate for a hostname. Or all routes should point to the same certificate.
We have the same problem in Ingress v1.

Reply:
👍 That's a great point, that seems like something we should also document on the Route hostnames field.

Reply:
Updated. PTAL.